core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -261,9 +268,19 @@ private String updateBuildConfig(OpenShiftClient client, String buildName, Build
         return buildName;
     }
 
-    private BuildStrategy createBuildStrategy(ImageConfiguration imageConfig, OpenShiftBuildStrategy osBuildStrategy) {
+    private BuildStrategy createBuildStrategy(ImageConfiguration imageConfig, OpenShiftBuildStrategy osBuildStrategy, String openshiftPullSecret) {
         if (osBuildStrategy == OpenShiftBuildStrategy.docker) {
-            return new BuildStrategyBuilder().withType("Docker").build();
+            BuildStrategy buildStrategy = new BuildStrategyBuilder().withType("Docker")
+                    .withNewDockerStrategy()
+                    .endDockerStrategy().build();
+
+            if (openshiftPullSecret != null) {
+               buildStrategy.setDockerStrategy(new DockerBuildStrategyBuilder()
```

Comment:
Would it be better to refactor this code as ?

```
buildStrategy.getDockerStrategy().setPullSecret(
      new LocalObjectReferenceBuilder()
               .withName(openshiftPullSecret)
               .build()
);
````

It would preserve other docker startegy attributes rather tha override it completely.